cirq-core/cirq/circuits/moment.py

```diff
@@ -223,7 +226,7 @@ def with_operations(self, *contents: 'cirq.OP_TREE') -> 'cirq.Moment':
         m._qubit_to_op = self._qubit_to_op.copy()
         qubits = set(self._qubits)
         for op in flattened_contents:
-            if any(q in qubits for q in op.qubits):
+            if _compat.__cirq_debug__.get() and any(q in qubits for q in op.qubits):
```

Comment:
nit: Move the `if _compat.__cirq_debug__.get()` outside of the `for op in flattened_contents:` loop so that the `__cirq_debug__` condition is executed only once